src/syntax/syntax.rkt

```diff
@@ -238,101 +235,93 @@
-  (define-operator (name real) real [ival ival-impl]))
-
-(define-syntax-rule (define-2ary-real-operator name ival-impl)
-  (define-operator (name real real) real [ival ival-impl]))
-
-(define-syntax-rule (define-1ary-real-operators [name ival-impl] ...)
-  (begin
-    (define-1ary-real-operator name ival-impl) ...))
-
-(define-syntax-rule (define-2ary-real-operators [name ival-impl] ...)
-  (begin
-    (define-2ary-real-operator name ival-impl) ...))
-
-(define-1ary-real-operators [neg ival-neg]
-                            [acos ival-acos]
-                            [acosh ival-acosh]
-                            [asin ival-asin]
-                            [asinh ival-asinh]
-                            [atan ival-atan]
-                            [atanh ival-atanh]
-                            [cbrt ival-cbrt]
-                            [ceil ival-ceil]
-                            [cos ival-cos]
-                            [cosh ival-cosh]
-                            [erf ival-erf]
-                            [exp ival-exp]
-                            [exp2 ival-exp2]
-                            [fabs ival-fabs]
-                            [floor ival-floor]
-                            [lgamma ival-lgamma]
-                            [log ival-log]
-                            [log10 ival-log10]
-                            [log2 ival-log2]
-                            [logb ival-logb]
-                            [rint ival-rint]
-                            [round ival-round]
-                            [sin ival-sin]
-                            [sinh ival-sinh]
-                            [sqrt ival-sqrt]
-                            [tan ival-tan]
-                            [tanh ival-tanh]
-                            [tgamma ival-tgamma]
-                            [trunc ival-trunc])
-
-(define-2ary-real-operators [+ ival-add]
-                            [- ival-sub]
-                            [* ival-mult]
-                            [/ ival-div]
-                            [atan2 ival-atan2]
-                            [copysign ival-copysign]
-                            [fdim ival-fdim]
-                            [fmax ival-fmax]
-                            [fmin ival-fmin]
-                            [fmod ival-fmod]
-                            [pow ival-pow]
-                            [remainder ival-remainder])
-
-(define-operator (== real real) bool [ival ival-==])
-
-(define-operator (!= real real) bool [ival ival-!=])
-
-(define-operator (< real real) bool [ival ival-<])
-
-(define-operator (> real real) bool [ival ival->])
-
-(define-operator (<= real real) bool [ival ival-<=])
-
-(define-operator (>= real real) bool [ival ival->=])
-
-;; logical operators ;;
-
-(define-operator (not bool) bool [ival ival-not])
+(define-syntax define-operators
+  (syntax-rules (:)
+    [(_ [name : itype ... otype] ...)
+     (begin
+       (define-operator (name itype ...) otype) ...)]))
```

Comment:
I think these should have a spec of `(lambda (x) (name x))` and then every operator will have a spec.